parsl/tests/test_python_apps/test_join.py

```diff
@@ -1,17 +1,24 @@
 import pytest
-import time
 
 from parsl import join_app, python_app
 from parsl.dataflow.errors import JoinError
 
-from parsl.tests.configs.local_threads import fresh_config as local_config
+
+class _InnerAppCalledMoreThanOnce(Exception):
+    pass
+
 
 RESULT_CONSTANT = 3
 
+_inner_app_called = False
+
 
 @python_app(cache=True)
 def inner_app():
-    time.sleep(1)
+    global _inner_app_called
+    if _inner_app_called:
+        raise _InnerAppCalledMoreThanOnce()
+    _inner_app_called = True
```

Comment:
probably a tight race condition here if two worker threads launch inner app very close together in time?

Reply:
*nod; now that I look at the tests, I don't see that they care about the `cache=True` aspect at all.  Removing that logic entirely in lieu of another test that specifically tests the caching.